src/main/java/com/ubc/cpsc304/hotelify/controller/CustomerController.java

```diff
@@ -42,11 +51,62 @@ private CustomerResponseDto createCustomer(@RequestBody CustomerRequestDto custo
         return CustomerController.convertModel(createdCustomer);
     }
 
+    @GetMapping("/{username}")
+    private CustomerResponseDto getCustomer(@PathVariable String username)
+            throws BadRequestException {
+
+        Customer customer = this.customerService.findByUsername(username);
+
+        return CustomerController.convertModel(customer);
+    }
+
+    @PostMapping("/{username}/address")
```

Comment:
I actually like `("/address/{username}"` better

Reply:
In my understand:
user **has** address. so the address is under user URL

Reply:
But it is POST /customers/{username}/address. 
So I think address is under customers, and {username} should be used to specify which user is it for this operation.